java/core/src/java/org/apache/orc/impl/RecordReaderUtils.java

```diff
@@ -131,11 +131,21 @@ public boolean isTrackingDiskRanges() {
       return zcr != null;
     }
 
+    /**
+     * @deprecated Use {@link #releaseAllBuffers()} instead. This method was
+     * incorrectly added and shouldn't be used anymore.
+     */
+    @Deprecated
     @Override
     public void releaseBuffer(ByteBuffer buffer) {
```

Comment:
After this PR, Apache ORC doesn't use this code path, right?

Reply:
Yes

Reply:
External systems may use this interface, e.g. hive llap, could you reconfirm that this.

Reply:
I'm sorry for later response, I missed this comment...
I checked hive llap codes, it directly uses the original buffer of zrc `readBuffer` returning to create `BufferChunk`, so there seems to be no problem.
I think if external systems use this interface directly, they should trace the bytebuff of zrc `readBuffer` returning by themselves, then realease it.
In previous ORC code, we don't trace the original zrc returning bytebuffer, so it is buggy.
WDYT, thanks for your review and look forward to your reply.